RyuSocks/Packets/UaPPacket.cs

```diff
@@ -0,0 +1,21 @@
+namespace RyuSocks.Packets
+{
+    public struct UaPPacket(string username, string password) : IAuthenticationPacket
+    {
+        public byte Version { get; set; }
+        public int UsernameLength = username.Length;
```

Comment:
The RFC specifies that this field is 1 byte long, so you should be using `byte` instead of `int`.